src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -368,34 +312,42 @@ private static object BindToCollection(Type type, IConfiguration config, BinderO
                         return;
                     }
 
-                    object? boundFromInterface = AttemptBindToCollectionInterfaces(type, config, options);
-                    if (boundFromInterface != null)
+                    // For other mutable interfaces like ICollection<> and ISet<>, we prefer copying values and setting them
+                    // on a new instance of the interface over populating the existing instance implementing the interface.
+                    // This has already been done, so there's not need to check again. For dictionaries, we fill the existing
+                    // instance if there is one (which hasn't happened yet), and only create a new instance if necessary.
+                    if (TypeIsADictionaryInterface(type))
                     {
-                        bindingPoint.SetValue(boundFromInterface);
-                        return; // We are already done if binding to a new collection instance worked
+                        Type typeOfKey = type.GenericTypeArguments[0];
+                        Type typeOfValue = type.GenericTypeArguments[1];
+                        // Overwrite type in case it was a IReadOnlyDictionary<>. We still want to be able to bind items.
+                        // REVIEW: What about settable IReadOnlyDictionary<> instances with an initial value?
+                        // I think we should consider preferring copying like we do for all other collection interfaces.
```

Comment:
What is the plan to address this comment?

Reply:
The last few commits is a first pass attempt at addressing this. Dictionary _interfaces_ are cloned and bound. 
_Concrete_ dictionaries are _not_ cloned, as this would be a breaking change to existing behaviour.
I've marked this as resolved, but I imagine there to be a few more comments re. the implementation.

Reply:
Can we remove the comment?